pkg/controller/ephemeralcluster/reconciler.go

```diff
@@ -258,6 +248,50 @@ func (r *reconciler) generateCIOperatorConfig(log *logrus.Entry, ec *ephemeralcl
 	}, nil
 }
 
+func (r *reconciler) handleCreateProwJob(ctx context.Context, log *logrus.Entry, ec *ephemeralclusterv1.EphemeralCluster, observedStatus *ephemeralclusterv1.EphemeralClusterStatus) (reconcile.Result, error) {
+	pjsForEC := prowv1.ProwJobList{}
+	listOpts := []ctrlruntimeclient.ListOption{
+		ctrlruntimeclient.MatchingLabels{EphemeralClusterLabel: ec.Name},
+		ctrlruntimeclient.InNamespace(r.prowConfigAgent.Config().ProwJobNamespace),
+	}
+	if err := r.masterClient.List(ctx, &pjsForEC, listOpts...); err != nil {
+		log.WithError(err).Error("Failed to list ProwJobs to check whether the request has one associated already")
+		return reconcile.Result{}, fmt.Errorf("list pjs to create ec: %w", err)
+	}
+
+	// This is a case that should never happen but if it does it means that either someone
+	// is creating PJs manually or there is a bug in this controller.
+	if len(pjsForEC.Items) > 1 {
+		log.Error("Too many ProwJobs bound, this is a bug")
```

Comment:
A typo? Should it be `found` instead of `bound` or `bounded`?

Reply:
`bound` is what I mean here. If we branch into that `if`, it means there is more than a single ProwJob associated (bound) to an EphemeralCluster, and this is an invalid condition, hence a bug in this controller.